pkg/network/dhcp/configurator.go

```diff
@@ -0,0 +1,72 @@
+package dhcp
+
+import (
+	"fmt"
+	"os"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
+)
+
+type Configurator struct {
+	advertisingIfaceName string
+	cacheFactory         cache.InterfaceCacheFactory
+	filterByMac          bool
+	handler              netdriver.NetworkHandler
+	launcherPID          string
+}
+
+// NewConfiguratorWithClientFilter should be used when the DHCP server is
+// expected to only reply to the MAC specified in the `cache.DhcpConfig` struct
+func NewConfiguratorWithClientFilter(cacheFactory cache.InterfaceCacheFactory, launcherPID string, advertisingIfaceName string, handler netdriver.NetworkHandler) *Configurator {
+	return &Configurator{
+		advertisingIfaceName: advertisingIfaceName,
+		cacheFactory:         cacheFactory,
+		launcherPID:          launcherPID,
+		filterByMac:          true,
+		handler:              handler,
+	}
+}
+
+// NewConfigurator should be used when the DHCP server is
+// expected to reply all client requests, independently of their MAC address
+func NewConfigurator(cacheFactory cache.InterfaceCacheFactory, launcherPID string, advertisingIfaceName string, handler netdriver.NetworkHandler) *Configurator {
+	return &Configurator{
+		advertisingIfaceName: advertisingIfaceName,
+		cacheFactory:         cacheFactory,
+		launcherPID:          launcherPID,
+		filterByMac:          false,
+		handler:              handler,
+	}
+}
+
+func (d Configurator) ImportConfiguration(ifaceName string) (*cache.DhcpConfig, error) {
+	dhcpConfig, err := d.cacheFactory.CacheDhcpConfigForPid(d.launcherPID).Read(ifaceName)
+	if err != nil {
+		return nil, err
+	}
+	dhcpConfig.AdvertisingIPAddr = dhcpConfig.AdvertisingIPAddr.To4()
+	dhcpConfig.AdvertisingIPv6Addr = dhcpConfig.AdvertisingIPv6Addr.To16()
+	return dhcpConfig, nil
+}
+
+func (d Configurator) ExportConfiguration(config cache.DhcpConfig) error {
+	return d.cacheFactory.CacheDhcpConfigForPid(d.launcherPID).Write(config.Name, &config)
+}
+
+func (d Configurator) EnsureDhcpServerStarted(podInterfaceName string, dhcpConfig cache.DhcpConfig, dhcpOptions *v1.DHCPOptions) error {
+	dhcpStartedFile := fmt.Sprintf("/var/run/kubevirt-private/dhcp_started-%s", podInterfaceName)
+	_, err := os.Stat(dhcpStartedFile)
+	if os.IsNotExist(err) && !dhcpConfig.IPAMDisabled {
```

Comment:
Just a nit: I would move the `dhcpConfig.IPAMDisabled` to the beginning of the method to avoid redundant io call.